pandas/core/generic.py

```diff
@@ -4287,6 +4287,38 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
         else:
             return result
 
+    def _clip_with_one_bound(self, threshold, method, axis, inplace):
+
+        if np.any(isnull(threshold)):
+            raise ValueError("Cannot use an NA value as a clip threshold")
+
+        # method is self.le for upper bound and self.ge for lower bound
+        if is_scalar(threshold) and is_number(threshold):
+            if method.__name__ == 'le':
+                return self._clip_with_scalar(None, threshold, inplace=inplace)
+            else:
+                return self._clip_with_scalar(threshold, None, inplace=inplace)
+
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+
+        subset = method(threshold, axis=axis) | isnull(self)
+
+        # GH #15390
+        if is_scalar(threshold) or is_number(threshold):
+            return self.where(subset, threshold, axis=axis, inplace=inplace)
+
```

Comment:
hmm, I think you should be able to do

``threshold = np.asarray(threshold)`` and this should just work; it will broadcast when comparing in where.